app/models/pillar.rb

```diff
@@ -2,12 +2,18 @@
 # rubocop:disable Metrics/ClassLength
 class Pillar < ApplicationRecord
   validates :pillar, presence: true
-  validates :value, presence: true
+  validates :value, presence: true, eviction: true
 
   scope :global, -> { where minion_id: nil }
 
   PROTECTED_PILLARS = [:dashboard, :apiserver, :dashboard_external_fqdn].freeze
 
+  # update_or_remove! updates the current pillar with the given value. That
+  # being said, if the given value is blank, then it will destroy this pillar.
+  def update_or_remove!(val)
+    val.blank? ? destroy : update_attributes!(value: val)
+  end
```

Comment:
Isn't this method overlapping logic with https://github.com/mssola/velum/blob/ce4da5af9c710cf4068985a3b1b69488936e3a9f/app/models/pillar.rb#L149-L150?

Reply:
Not exactly. This one will always raise an exception on error, while the other simply calls `save` which does not raise an exception. This is important because this way we'll rollback the transaction on the controller.

Reply:
Thank you!